lib/combinePrimitives.js

```diff
@@ -0,0 +1,270 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+var DeveloperError = Cesium.DeveloperError;
+var loadGltfUris = require('./loadGltfUris');
+module.exports = combinePrimitives;
+
+//Combines primitives in a mesh which have the same material, mode, and attributes.
+function combinePrimitives(gltf) {
+    if (defined(gltf.accessors) && defined(gltf.bufferViews) && defined(gltf.buffers)) {
+        loadGltfUris(gltf);
+        var meshes = gltf.meshes;
+        if (defined(meshes)) {
+            for (var meshId in meshes) {
+                if (meshes.hasOwnProperty(meshId)) {
+                    var mesh = meshes[meshId];
+                    var primitives = mesh.primitives;
+                    if (defined(primitives)) {
+                        mesh.primitives = combineMeshPrimitives(gltf, meshId, primitives);
+                    }
+                }
+            }
+        }
+    }
+
+    return gltf;
+}
+
+//Combines the primitives for a given mesh.
+function combineMeshPrimitives(gltf, meshId, primitives) {
+    //Group the primitives which share the same material, mode, and attibutes.
+    var primitiveGroups = [];
+    var combinedPrimitives = [];
+    while (primitives.length > 0) {
+        var currentPrimitive = primitives.shift();
+        var currentGroup = [currentPrimitive];
+        for (var i = 0; i < primitives.length; i++) {
+            var nextPrimitive = primitives[i];
+            if (readyToCombine(currentPrimitive, nextPrimitive)) {
+                currentGroup.push(nextPrimitive);
+                primitives.splice(i, 1);
+                i--;
+            }
+        }
+        primitiveGroups.push(currentGroup);
+    }
+
+    for (var j = 0; j < primitiveGroups.length; j++) {
+        combinedPrimitives.push(combinePrimitiveGroup(gltf, meshId, primitiveGroups[j]));
+    }
+
+    return combinedPrimitives;
+}
+
+
+//Checks if two primitives can be combined based on their attributes, materials, and modes.
+function readyToCombine(a, b) {
+    if (a.material !== b.material || a.mode !== b.mode || defined(a.indices) !== defined(b.indices)) {

```

Comment:
Ah, I didn't catch the `defined` call.  You are right.  Both should have indices.  We could consider adding indices if one primitive has indices and the other does not, but this would also need heuristics to see when it is worth it.  No need to consider this until we know if it is an issue in practice.